go-controller/pkg/ovn/endpoints.go

```diff
@@ -268,6 +254,10 @@ func (ovn *Controller) deleteEndpoints(ep *kapi.Endpoints) error {
 		}
 		vip := util.JoinHostPortInt32(svc.Spec.ClusterIP, svcPort.Port)
 		ovn.removeServiceEndpoints(lb, vip)
+
+		if util.ServiceTypeHasNodePort(svc) {
+			ovn.deleteGatewayVIPs(svcPort.Protocol, svcPort.NodePort)
```

Comment:
What is this? Is this a bugfix? It should go in a separate commit from the cleanups. (And if one of the new tests is to to test this fix, that new test should go in the commit with this fix.)

Reply:
Ok, will do.